lib/kubernetes-deploy/kubernetes_resource.rb

```diff
@@ -31,7 +31,9 @@ class << self
       def build(namespace:, context:, definition:, logger:, statsd_tags:)
         opts = { namespace: namespace, context: context, definition: definition, logger: logger,
                  statsd_tags: statsd_tags }
-        if KubernetesDeploy.const_defined?(definition["kind"])
+        if definition["kind"].blank?
+          raise InvalidTemplateError.new("Template missing 'Kind'", filename: nil, content: definition.to_s)
```

Comment:
Since we have a valid case for not requiring `filename` now, let's make the default nil in the initializer so you can leave it off here.

Reply:
I think we should `to_yaml` the definition instead of `to_s` it, for consistency with the other errors (and readability when the definition is long). Compare:

![image](https://user-images.githubusercontent.com/4789493/38948137-a1b267ec-430c-11e8-9e9e-102f372ae8db.png)


Versus `test_invalid_yaml_fails_fast`:

![image](https://user-images.githubusercontent.com/4789493/38948119-9619086e-430c-11e8-9aeb-f6c74b25751f.png)